### PR TITLE
Log not found equipment

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests: NullPointerException on a n-1 contingency with an equipment id not found

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 1]
```diff
@@ -186,15 +192,30 @@ private SecurityAnalysisResult run(SecurityAnalysisRunContext context, UUID resu
...
         Reporter rootReporter = Reporter.NO_OP;
         Reporter reporter = Reporter.NO_OP;
+        Reporter elementNotFoundSubReporter;
+        List<Report> notFoundElementReports = new ArrayList<>();
+        Map<String, String> notFE = new HashMap<>();
+        contingencies.stream()
+                .filter(contingencyInfos -> !CollectionUtils.isEmpty(contingencyInfos.getNotFoundElements()))
+                .forEach(contingencyInfos -> {
+                    String elementsIds = String.join(", ", contingencyInfos.getNotFoundElements());
+                    notFE.put(contingencyInfos.getContingency().getId(), elementsIds);
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java]: Done

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 2]
```diff
@@ -204,12 +225,25 @@ private SecurityAnalysisResult run(SecurityAnalysisRunContext context, UUID resu
...
+                reporter,
+                resultUuid);
 
         SecurityAnalysisResult result = future == null ? null : future.get();
         if (context.getReportUuid() != null) {
-            Reporter finalRootReporter = rootReporter;
-            reportService.sendReport(context.getReportUuid(), finalRootReporter);
+            if (!CollectionUtils.isEmpty(notFoundElementReports)) {
+                elementNotFoundSubReporter = reporter.createSubReporter(context.getReportUuid().toString() + "notFoundElements", "Elements not found");
+                elementNotFoundSubReporter.report(Report.builder()
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java]: Removed

**Review by FranckLecuyer — APPROVED** [review]
Code review: OK
Tests: OK

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 3]
```diff
@@ -186,15 +191,27 @@ private SecurityAnalysisResult run(SecurityAnalysisRunContext context, UUID resu
...
+        List<ContingencyInfos> contingencies = context.getContingencyListNames().stream()
+                .map(contingencyListName -> actionsService.getContingencyList(contingencyListName, context.getNetworkUuid(), context.getVariantId()))
+                .flatMap(List::stream)
+                .toList();
 
         SecurityAnalysis.Runner securityAnalysisRunner = securityAnalysisFactorySupplier.apply(context.getProvider());
 
         Reporter rootReporter = Reporter.NO_OP;
         Reporter reporter = Reporter.NO_OP;
+        List<Report> notFoundElementReports = new ArrayList<>();
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java:203]: You can move this to row 235

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 4]
```diff
@@ -204,12 +221,23 @@ private SecurityAnalysisResult run(SecurityAnalysisRunContext context, UUID resu
...
         }
 
-        CompletableFuture<SecurityAnalysisResult> future = runASAsync(context, securityAnalysisRunner, network, contingencies, reporter, resultUuid);
+        CompletableFuture<SecurityAnalysisResult> future = runASAsync(context,
+                securityAnalysisRunner,
+                network,
+                contingencies.stream()
+                        .map(ContingencyInfos::getContingency)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList()),
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java:230]: .filter(Objects::nonNull)
                        .toList())

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**